### ISSUE TITLE
Add argument for build name

### ISSUE THREAD

**Issue by westarne** [jenkinsci/stashnotifier-plugin#241]
An argument to optionally override the build name send in the status notification would be great to make it possible to send one status per build step. This should result in a bitbucket display like this:

![image](https://user-images.githubusercontent.com/15111918/62541095-5d64d480-b859-11e9-86c7-92796592d249.png)

Example usage:
`notifyBitbucket(buildName: "Build")`
Without this parameter I would expect the build name to be derived like before.

**Comment by scaytrase** [comment]
I'm not really sure how it should behave in terms of generating links, etc.

**Comment by westarne** [comment]
I'd expect the plugin to link like before. I'm not sure how the links are generated right now, but all of those entries should generally link to the same build. If possible it could link to the exact build step (which is possible as far as i know for the Blue Ocean Pipeline UI).

**Comment by scaytrase** [comment]
If you don't care about links you can do this kind of setup by yourself by manipulating project key and description while performing notification. different project key will result in different build records

**Comment by westarne** [comment]
Thank you for the advice. I think this will work as a workaround. But in the long term i would look forward to a direct parameter without the need to change build parameters back and forth.

Edit: Actually only the `displayName` can be overwritten. So it is not possible to fully customize the build name this way.

**Comment by westarne** [comment]
Also it is not possible to overwrite `currentBuild.result` multiple times:

```groovy
        currentBuild.result = 'SUCCESS'
        echo currentBuild.result
        currentBuild.result = 'NOT_BUILD'
        echo currentBuild.result
        currentBuild.result = 'SUCCESS'
        echo currentBuild.result
        currentBuild.result = 'NOT_BUILD'
        echo currentBuild.result
        currentBuild.result = 'FAILURE'
        echo currentBuild.result
        currentBuild.result = 'UNSTABLE'
        echo currentBuild.result
        currentBuild.result = 'FAILURE'
        echo currentBuild.result
```

will produce:

```
[Pipeline] echo
SUCCESS
[Pipeline] echo
FAILURE
[Pipeline] echo
FAILURE
[Pipeline] echo
FAILURE
[Pipeline] echo
FAILURE
[Pipeline] echo
FAILURE
[Pipeline] echo
FAILURE
```

This is part of the https://github.com/jenkinsci/stashnotifier-plugin/issues/209 issue.

**Comment by scaytrase** [comment]
@geralvin @westarne I would appreciate a PR for (buildName+buildKey) and\or buildState management with a BC keep. Why not if it solves user problems

**Comment by geralvin** [comment]
what is BC ?

Anyway I'll try make some changes.

**Comment by scaytrase** [comment]
Backward compatibility

**Comment by westarne** [comment]
I've added a merge request https://github.com/jenkinsci/stashnotifier-plugin/pull/243

**Comment by scaytrase** [comment]
Fixed in #243
